src/pseudopeople/dataset.py

```diff
@@ -165,6 +165,12 @@ def _reformat_dates_for_noising(self) -> None:
 
         self.data = data
 
+    @staticmethod
+    def drop_non_schema_columns(
+        data: pd.DataFrame, dataset_schema: DatasetSchema
+    ) -> pd.DataFrame:
+        return data[[c.name for c in dataset_schema.columns]]
```

Comment:
Nit: add docstrings, incl parameters and returns. Seems silly since it's a one-liner but remember those docstrings get added to the readthedocs where you don't see the code itself.